trieste/models/gpflux/sampler.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import tensorflow as tf
+from gpflux.layers import GPLayer, LatentVariableLayer
+from gpflux.models import DeepGP
+from gpflux.sampling.sample import Sample
+
+from ...types import TensorType
+from ...utils import DEFAULTS
+from ..interfaces import (
+    ProbabilisticModel,
+    ReparametrizationSampler,
+    TrajectoryFunction,
+    TrajectorySampler,
+)
+from .interface import GPfluxPredictor
+
+
+def sample_consistent_lv_layer(layer: LatentVariableLayer) -> Sample:
+    class SampleLV(Sample):  # type: ignore[misc]
+        def __call__(self, X: TensorType) -> tf.Tensor:
+            sample = layer.prior.sample()
+            batch_shape = tf.shape(X)[:-1]
+            sample_rank = tf.rank(sample)
+            for _ in range(len(batch_shape)):
+                sample = tf.expand_dims(sample, 0)
+            sample = tf.tile(
+                sample, tf.concat([batch_shape, tf.ones(sample_rank, dtype="int32")], -1)
+            )
+            return layer.compositor([X, sample])
+
+    return SampleLV()
+
+
+def sample_dgp(model: DeepGP) -> TrajectoryFunction:
```

Comment:
how come this type of sampling is not in GPflux? this definitely looks like it should be part of GPflux...
(probably we went over this and I forgot about it in the meantime :) )

Reply:
also, docstring is missing

Reply:
For the first point, it is (in `GPflux/gpflux./models/deep_gp.py`), but it doesn't handle sampling from latent variable layers. Will add the docstring :)

Reply:
Also this does have the slight change that it returns a Trieste `TrajectoryFunction`, and not a GPflux `Sample`

Reply:
also, slightly changed name could be more precise `sample_deepGP`?